### PR TITLE
[wrdac/subSystem_ARE]: address #18

### PR DESCRIPTION
**Pull request by pattacini**
This PR should address #18.

Everything should still work as before if the attempt to connect to `depth2kin` is unsuccessful.
Further, if `depth2kin` is not calibrated, then it will return the same target fed in the request.

/cc @Tobias-Fischer

### CODE REVIEW

File: main/src/libraries/wrdac/include/wrdac/subsystems/subSystem_ARE.h [hunk 1]
```diff
@@ -49,21 +50,70 @@ namespace wysiwyd {
...
+            {
+                std::string hand="";
+                for (int i=0; i<options.size(); i++)
+                {
+                    yarp::os::Value val=options.get(i);
+                    if (val.isString())
+                    {
+                        std::string item=val.asString();
+                        if ((item=="left") || (item=="right"))
+                            hand=item;
```
Comment (Tobias-Fischer) [inline main/src/libraries/wrdac/include/wrdac/subsystems/subSystem_ARE.h]: Can you add a `break` here? :)

File: main/src/libraries/wrdac/include/wrdac/subsystems/subSystem_ARE.h [hunk 2]
```diff
@@ -49,21 +50,70 @@ namespace wysiwyd {
...
+                // always choose hand
+                if (hand.empty())
+                {
+                    if (handToUse.empty())
+                    {
+                        hand=(target[1]>0.0?"right":"left"); 
+                        options.addString(hand.c_str());
+                    }
+                    else
+                        hand=lastlyUsedHand;
```
Comment (Tobias-Fischer) [inline main/src/libraries/wrdac/include/wrdac/subsystems/subSystem_ARE.h]: Shouldn't this be `hand=handToUse`?

**Comment by Tobias-Fischer** [comment]
I don't get the logic behind the `lastlyUsedHand` (maybe it's just to late), could you please briefly explain? We'll do some tests early next week, and then merge.

File: main/src/libraries/wrdac/include/wrdac/subsystems/subSystem_ARE.h (hunk 2 above)
Comment (pattacini) [inline main/src/libraries/wrdac/include/wrdac/subsystems/subSystem_ARE.h]: Actually, `lastlyUsedHand` serves a different purpose: to know which was the hand used last time for checking the correct hand in `dropOn`.

**Comment by pattacini** [comment]
Yes, two shoots two kills.

**Comment by pattacini** [comment]
Btw, no hurry, I'm coding here at the Munich airport :smile:

**Comment by pattacini** [comment]
Be careful, history changed: branch just rebased on master.

**Comment by Tobias-Fischer** [comment]
@pattacini: Is it okay if we check this in Lyon in a couple of weeks?

**Comment by pattacini** [comment]
Yep :+1:

**Comment by Tobias-Fischer** [comment]
Probably deprecated, as we likely will rely on https://github.com/robotology/wysiwyd/pull/83. Will close as soon as https://github.com/robotology/wysiwyd/pull/83 is merged.

**Comment by pattacini** [comment]
Deprecated indeed.

**Pull request merged**